src/converters/app-converter.ts

```diff
@@ -0,0 +1,388 @@
+            }
+            return key;
+        }
+
+        this.parameters.set(logicalId, parameterValue(this.host));
+    }
+
+    private mapResource(
+        logicalId: string,
+        typeName: string,
+        props: any,
+        options: pulumi.ResourceOptions,
+    ): ResourceMapping {
+        if (this.host.options?.remapCloudControlResource !== undefined) {
+            const res = this.host.options.remapCloudControlResource(logicalId, typeName, props, options);
+            if (res !== undefined) {
+                debug(`remapped ${logicalId}`);
+                return res;
+            }
+        }
+
+        const awsMapping = mapToAwsResource(logicalId, typeName, props, options);
+        if (awsMapping !== undefined) {
+            debug(`mapped ${logicalId} to classic AWS resource(s)`);
+            return awsMapping;
+        }
+
+        return mapToCfnResource(logicalId, typeName, props, options);
+    }
+
+    private processOptions(resource: CloudFormationResource, parent: pulumi.Resource): pulumi.ResourceOptions {
+        const dependsOn = getDependsOn(resource);
+        return {
+            parent: parent,
+            dependsOn: dependsOn !== undefined ? dependsOn.map((id) => this.resources.get(id)!.resource) : undefined,
+        };
+    }
+
+    /** @internal */
+    asOutputValue<T>(v: T): T {
+        const value = this.host.stack.resolve(v);
+        return this.processIntrinsics(value) as T;
+    }
+
+    private processIntrinsics(obj: any): any {
+        debug(`Processing intrinsics for ${JSON.stringify(obj)}`);
+        if (typeof obj === 'string') {
+            return obj;
+        }
+
+        if (typeof obj !== 'object') {
+            return obj;
+        }
+
+        if (Array.isArray(obj)) {
+            return obj.filter((x) => !this.isNoValue(x)).map((x) => this.processIntrinsics(x));
+        }
+
+        const ref = obj.Ref;
+        if (ref) {
+            return this.resolveRef(ref);
+        }
+
+        const keys = Object.keys(obj);
+        if (keys.length == 1 && keys[0]?.startsWith('Fn::')) {
+            return this.resolveIntrinsic(keys[0], obj[keys[0]]);
+        }
+
+        return Object.entries(obj)
+            .filter(([_, v]) => !this.isNoValue(v))
+            .reduce((result, [k, v]) => ({ ...result, [k]: this.processIntrinsics(v) }), {});
+    }
+
+    private isNoValue(obj: any): boolean {
+        return obj?.Ref === 'AWS::NoValue';
+    }
+
+    private resolveOutput(repr: OutputRepr): pulumi.Output<any> {
+        return OutputMap.instance().lookupOutput(repr)!;
+    }
+
+    private resolveIntrinsic(fn: string, params: any) {
+        switch (fn) {
+            case 'Fn::GetAtt': {
+                debug(`Fn::GetAtt(${params[0]}, ${params[1]})`);
+                return this.resolveAtt(params[0], params[1]);
+            }
+
+            case 'Fn::Join':
+                return lift(([delim, strings]) => strings.join(delim), this.processIntrinsics(params));
```

Comment:
Powerful meta-programming here. Again I'd like to discuss a bit other representations then `any` if they add any value for us here.

Reply:
Yes, I would like to refactor this whole `lift` thing. It is very hard to track and I already fixed one bug where `Fn::Base64` wasn't even working.